### ISSUE TITLE
Dropdown under footer

### ISSUE THREAD

**Issue by Iliasvw** [JedWatson/react-select#3263]
Hello everybody

I'm currently facing an issue when using react-select. 

![image](https://user-images.githubusercontent.com/17167673/49731833-e6048400-fc7c-11e8-954c-6e9acf8ed3e8.png)

As you can see, when I open the select, the dropdown is partly displayed under the footer. I have tried everything (z-index, overflow, position, ...).

The current style applied to react-select:
```
* {
    cursor: default !important;
    font-family: 'Open sans', sans-serif !important;
    outline: none !important;
  }

  .Select-control {
    background-color: ${({ theme }) => theme.colors.lightText};
    border-radius: 0.375rem;
    border: 0;
    box-shadow: 0.0625rem 0.0625rem 0.1875rem rgba(0,0,0,0.3) inset;
    color: ${({ theme }) => theme.colors.accentHover};
    font-size: 0.8rem;
    font-weight: normal;
    height: 2.5rem;

    ${core.mediaQueries.medium`
      font-size: 0.9rem;
    `};
  }

  .is-open > .Select-control {
    background-color: ${({ theme }) => theme.colors.lightText};
  }

  .Select-control:hover {
    box-shadow: 0.0625rem 0.0625rem 0.1875rem rgba(0,0,0,0.3) inset;
  }

  .Select-arrow-zone {
    align-items: center;
    display: flex;
    justify-content: center;
  }

  .Select-control .Select-input:focus {
    outline: none;
    outline-offset: 0;
  }

  .Select-control .Select-value {
    height: 100%;
  }

  .Select-control .Select-value .Select-value-label {
    height: 100%;
  }

  .Select-control .Select-value > div > div {
    margin-left: 1.5rem;
    width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    ${core.mediaQueries.medium`
      max-width: 95%;
      width: auto;
    `};
  }

  .Select-menu-outer > div > div > span {
    height: 0;
    margin: 0 1.5rem 0 0;
    width: 0;
  }

  .Select-menu-outer {
    max-height: 600px;
    overflow: hidden;

    ${core.mediaQueries.medium`
      max-height: 600px;
    `};

  }

  .Select-menu {
    max-height: 600px;
    overflow-y: visible !important;
  }

  .Select-menu-outer > div > div {
    font-size: 0.8rem;
    padding: 0.5rem 0;

    ${core.mediaQueries.medium`
      font-size: 0.9rem;
    `};

    &:last-child {
      margin-bottom: 2.5rem;
    }
  }

  .Select-menu-outer > div > div:hover {
    background-color: ${({ theme }) => theme.colors.lightText};
  }

  .is-focused:not(.is-open) > .Select-control, .is-focused.is-open > .Select-control {
    outline: none;
    outline-offset: 0;
  }

  .Select-noresults {
    margin-left: 1.5rem;
  }
```

When I change the z-index of the footer to -1, the dropdown is displayed good, but then the links in my footer aren't clickable...

Thanks in advance
Ilias

**Comment by Rall3n** [comment]
If you are using v2 of react-select, you can use the [`menuPortalTarget` prop](https://react-select.com/advanced#portaling) to render the dropdown into a parent container (e.g. `body`) of the footer.

```jsx
<Select
    { ... }
    menuPortalTarget={document.querySelector('body')}
/>
```
[CodeSandbox](https://codesandbox.io/s/m5o25j4k08)

**Comment by Iliasvw** [comment]
My package-lock.json says I'm using v1. 
=> "react-select": "^1.0.0-rc.10"
Any fix for this version?

**Comment by ebonow** [comment]
v1 is depreciated and no longer being worked but this should help you get up and running on the latest version:

https://react-select.com/upgrade-guide

**Comment by strap8** [comment]
const selectStyles = {
  container: (base, state) => ({
    ...base,
    zIndex: "999"
  })
};

**Comment by ClementeSerrano** [comment]
Any update on this issue? Im having the same problem

**Comment by 1nfility** [comment]
Solution for having a fixed dropdown menu, and keeping the width.

```
constructor(props) {
    super(props);

    this.div = React.createRef();
  }

  render() {
    return (
      <div style={{ width: '100%' }} ref={this.div}>
        <ReactSelect
          styles={{
            menu: (base) => ({
              position: 'fixed',
              width: this.div.current.offsetWidth,
              zIndex: 500,
              ...base
            })
          }}
          placeholder={'Test select'}
          options={[{ value: 'test', label: 'Test' }]}
        />
      </div>
    );
  }
```

**Comment by itsmeganesh-cse-iiit** [comment]
Same issue here , I have tried above solutions, but issue still there

**Comment by flexdinesh** [comment]
Hey y'all! As @Rall3n mentioned, there are ways to customize the looks in v2. If you're using v1, we recommend you to [upgrade to v2](https://react-select.com/upgrade-guide) as v1 is not maintained anymore. 

I'm closing this issue now. Pl feel free to reopen it if you have further questions in v2.

**Comment by flexdinesh** [comment]
If you're having this issue in v2 or later, pl follow the portalling instructions here — https://react-select.com/advanced#portaling

TLDR — you need these two additional props
```
menuPortalTarget={document.body}
styles={{ menuPortal: base => ({ ...base, zIndex: 9999 }) }}
```

**Comment by bladey** [comment]
Hi all,

In an effort to sustain the `react-select` project going forward, we're closing issues that appear to have been resolved via community comments or issues that are now redundant based on their age.

However, if you feel this issue is still relevant and you'd like us to review it - please leave a comment and we'll do our best to get back to you, and we'll re-open it if necessary.

**Comment by garrett-gottlieb** [comment]
Using @flexdinesh's solution works 50% of the time. Sometimes the overflow is cut off, other times it isn't and pushes more padding to the bottom of the body to make room for the menu. It's incredibly hard to debug, too, because clicking anywhere, including the pause button in chrome's debugger, causes the menu to disappear 🤔

**Comment by ebonow** [comment]
@garrett-gottlieb , `menuIsOpen` makes life much easier. You can either set it to true as a prop, or go into the dev react tools and set it to true for the StateManager.

**Comment by garrett-gottlieb** [comment]
@ebonow thanks for the advice. It turned out to be the `max-height: 27px`. How is this calculated?

I've set a min of 200 to fix:
```ts
  menuList: base => ({
    ...base,
    maxHeight: Math.max(
      typeof base.maxHeight === 'number' ? (base.maxHeight as number) : 0,
      200
    ),
  }),
```

**Comment by ebonow** [comment]
@garrett-gottlieb 
The maxHeight is computed by calculating the available space. Here is the code:
https://github.com/JedWatson/react-select/blob/b0411ff46bc1ecf45d9bca4fb58fbce1e57f847c/packages/react-select/src/components/Menu.js#L51

**Comment by johnbrittocred** [comment]
`onMenuOpen={() => window.scrollTo(0, document.body.scrollHeight)}`

this will do the trick if you have the dropdown in the bottom of page

**Comment by alcidesbsilvaneto** [comment]
> `onMenuOpen={() => window.scrollTo(0, document.body.scrollHeight)}`
> 
> this will do the trick if you have the dropdown in the bottom of page

Thanks, saved me some time. 
For those who have some fixed or absolute top-bar and it's desappearing with this solution, just add

`onMenuClose={() => window.scrollTo(0, 0)}`

**Comment by RichMatthews** [comment]
`menuPortalTarget={document.body}` works fine for desktop but we found it doesn't work on ipad or mobile. z-index alone also doesn't work here. does anyone have a solution for mobile?
